### PR TITLE
Add integration tests for the devfile registry

### PR DESCRIPTION
**Pull request by johnmcollier**
Signed-off-by: John Collier <jcollier@redhat.com>

**What does does this PR do / why we need it**:
This PR adds integration tests for the OCI-based devfile registry service. The tests are written using the Ginkgo framework and test the following:
- Index server endpoints (`/`, `/index`, `/index/sample`, `/index/all`, and `/devfile`)
- OCI registry endpoints and proxy
- Registry library

It also adds a Dockerfile for containerizing the test (as needed for AppSRE) and corresponding GitHub Action workflows to build/push the image to quay.io.

At this moment, these tests do not run as part of pull requests on the PRs and need to be run manually. A separate PR will enable these tests on PRs.

**Which issue(s) this PR fixes**:

https://github.com/devfile/api/issues/463 (partially)

**PR acceptance criteria**:

- [x] Test

- [x] Documentation

**How to test changes / Special notes to the reviewer**:
1. Checkout my branch
2. Run `./docker-build.sh` to build the test container
3. Run `export REGISTRY=https://registry.devfile.io`
4. Run `docker run devfile-registry-integration`

You should get output like the following:
```
johncollier@jcollier-mac integration % export REGISTRY=https://registry.devfile.io
johncollier@jcollier-mac integration % ./devfileregistry-integration
Running Devfile Registry integration tests...
Running Suite: Devfile Registry Tests
=====================================
Random Seed: 1621550347
Will run 11 of 11 specs

Starting to setup objects before run ginkgo suite
•••••••••••
Ran 11 of 11 Specs in 3.026 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

### CODE REVIEW

File: tests/integration/pkg/tests/devfilelibrary_tests.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/devfile/registry-support/tests/integration/pkg/util"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+// Integration/e2e test logic based on https://github.com/devfile/registry-operator/tree/master/test/integration
+// Tests use the CLI version of the registry-library to test.
+// Note: Requires adding the CLI to the system path before running these tests
+var _ = ginkgo.Describe("[Verify registry library works with registry]", func() {
```
Comment (GeekArthur) [inline tests/integration/pkg/tests/devfilelibrary_tests.go:30]: We can also test the case where listing both stack and sample, which to test `registry-library list --type all`.

**Review by GeekArthur — CHANGES_REQUESTED** [review]
It's great work, thanks for enabling the integration tests for OCI-based registry. I left one comment.

File: tests/integration/pkg/tests/devfilelibrary_tests.go (hunk 1 above)
Comment (johnmcollier) [reply tests/integration/pkg/tests/devfilelibrary_tests.go:30]: Thanks, good catch. Updated.

**Review by GeekArthur — APPROVED** [review]

**Comment by johnmcollier** [comment]
^ Had to fix permissions on the docker_push.sh script

**Review by GeekArthur — APPROVED** [review]

**Pull request merged**